### PR TITLE
adding ledgerbackend datastore txmeta as a data source

### PR DESCRIPTION
**Pull request by chowbao**
Integrate cdp changes to ingest from datastore txmeta files
* Added new option to read/create ledgerbackend using the ledgerbackend.datastore to interface with the txmeta files from gcs
* Controlled by two new parameters
  * captive-core - will use captive core; datastore gcs is the default for stellar-etl now
  * datastore-url - the url to the datastore (gcs bucket)
* Most export_* could use the LedgerCloseMeta to transform/process the data needed
* assets and ledgers which originally pulled from history archives had a custom xdr format that needed to be manually constructed from the LedgerCloseMeta
  * TODO: In the future could consider removing this dependency all together and use straight LedgerCloseMeta instead (different PR)
* TODO: Probably should delete all the unused from_genesis export_*s (different PR)
* TODO: All the parameters are kinda a mess; could be refactored (different PR)
* TODO: The export_* commands could also be refactored (different PR)

### CODE REVIEW

File: cmd/export_ledger_entry_changes.go
Comment (sydneynotthecity) [inline cmd/export_ledger_entry_changes.go:53]: lol

File: internal/input/assets_history_archive.go
Comment (sydneynotthecity) [inline internal/input/assets_history_archive.go:13]: even though the backend is different, might be nice to maintain parity between `GetPaymentOperations` and `GetPaymentOperationsHistoryArchive`. ie, maybe keep the input parameters for the methods consistent.

File: internal/input/ledgers_history_archive.go
Comment (sydneynotthecity) [inline internal/input/ledgers_history_archive.go:12]: same feedback as `GetPaymentOperation`

File: internal/input/operations.go
Comment (sydneynotthecity) [inline internal/input/operations.go:47]: nice, this is much cleaner

File: internal/utils/main.go
Comment (sydneynotthecity) [inline internal/utils/main.go:758]: is this method going to be agnostic to cloud provider? or will each cloud provider have its own backend?

File: cmd/export_assets.go
Comment (sydneynotthecity) [inline cmd/export_assets.go:33]: I like your idea of eventually refactoring this History Archives method to use lcm directly. I think it's cleaner and less confusing

**Review by sydneynotthecity — COMMENTED** [review]
Great start for ETL refactor. Small nit to get the unit test passing for `operation_trace_code` again.

I agree with all of your TODOs listed and think we should document them in tickets to be working in an upcoming sprint. 

Did you test the code using the `captive_core` flag to confirm that the code is backwards compatible?

File: internal/utils/main.go
Comment (chowbao) [reply internal/utils/main.go:758]: Yes it is cloud provider agnostic. It's just only the GCS datastore is defined right now

**Comment by chowbao** [comment]
Before merging:
* Wait for ledgerbackend to be merged and use a non-branch go repo https://github.com/stellar/go/pull/5260
* Create and point to a prod gcs bucket for txmeta instead of the ledgerexporter-test in dev

File: internal/utils/main.go
Comment (chowbao) [inline internal/utils/main.go:775]: This function was removed from the ingest library. Needed for account signers

**Review by sydneynotthecity — APPROVED** [review]
LGTM

**Pull request merged**
